### PR TITLE
Redesign initial upload page

### PR DESCRIPTION
**Pull request by lukenewman**
## Background
This PR redesigns the initial upload page.

## GitHub Issue
https://github.com/ctoec/data-collection/issues/1380

## Associated PRs
https://github.com/ctoec/component-library/pull/170

## Validation Plan
* head to the upload page
* ensure designs match design specs
* ensure file upload still works

## Automated Testing
- [ ] All unit tests are passing.
- [ ] If applicable, unit tests have been added to cover this changeset.

## Additional Context
N/A

### CODE REVIEW

File: client/src/containers/Upload/Upload.tsx [hunk 1]
```diff
@@ -108,15 +107,15 @@ const Upload: React.FC<StepContentProps> = ({
   }
 
   return (
-    <div className="grid-container">
-      <BackButton location="/" />
+    <div className="upload-container">
```
Comment (bamader) [inline client/src/containers/Upload/Upload.tsx:110]: All for the defining of a custom CSS class here to capture design changes, but I _think_ `grid-container` has specific attributes and properties for line heights, indents, spacing, etc. that makes it play nicely with grid rows later on. How  do we feel about having `div className="grid-container upload-container"` in the same line?

File: client/src/containers/Upload/UploadWizard.tsx [hunk 2]
```diff
@@ -49,35 +50,30 @@ const UploadWizard: React.FC = () => {
...
-            &nbsp;to enter all of the required state-funded enrollment data.
-            When you’re finished, you can upload the complete template below.
-            After your data is in the system, you can also{' '}
-            <Link to="/roster">
-              use the Roster to make changes to your data
-            </Link>
-            .
-          </p>
+          <div className="instructions-header">
+            <div className="margin-right-10">
```
Comment (bamader) [inline client/src/containers/Upload/UploadWizard.tsx:57]: This seems excessive? We've never had to use a margin of 10 anywhere in the app. Have you tried putting this all in a  `grid-container` div? That carries some automatic margining with it.

File: client/src/containers/Upload/UploadWizard.tsx [hunk 3]
```diff
@@ -49,35 +50,30 @@ const UploadWizard: React.FC = () => {
...
+              <h1 ref={h1Ref} className="margin-bottom-4 margin-top-6">
+                Upload your enrollment data
+              </h1>
+              <p>
+                Once you've filled out the{' '}
+                <Link to="/template">data collection template</Link>
+                &nbsp;— with all of the state funded enrollment required — you
```
Comment (bamader) [inline client/src/containers/Upload/UploadWizard.tsx:64]: Nit, typo: we want "...with all of the state-funded enrollment _data_ required" (data highlighted for emphasis, as it's currently missing).

**Review by bamader — APPROVED** [review]
LGTM aside from some minor margin things!

**Review by vonbearshark — APPROVED** [review]
Agreed with @bamader's feedback, but otherwise lgtm!

**Comment by JordanGuinn** [comment]
Looks like tests are failing?  I imagine the snapshots just need to be updated

File: client/src/containers/Upload/UploadWizard.tsx (hunk 2 above)
Comment (JordanGuinn) [reply client/src/containers/Upload/UploadWizard.tsx:57]: Agreed - some combination of [grid rows and grid columns](https://designsystem.digital.gov/utilities/layout-grid/) should be able to accomplish this same thing, while also maintaining a reasonable layout for the page.

**Review by JordanGuinn — APPROVED** [review]
Agreed with the sentiments in others' feedback, but beyond that LGTM 💯

File: client/src/containers/Upload/UploadWizard.tsx (hunk 2 above)
Comment (lukenewman) [reply client/src/containers/Upload/UploadWizard.tsx:57]: hmmmm not sure how to get inter-column margins right

**Comment by lukenewman** [comment]
looks like circleci can't install yarn packages?

**Comment by JordanGuinn** [comment]
Try updating both of your branches

**Comment by lukenewman** [comment]
updating how?

**Comment by JordanGuinn** [comment]
Merge in `base` everywhere, then repin `component-library` here.  Your lockfile looks to be out of date

**Comment by bamader** [comment]
I think he means pulling in the latest version of base, doing fresh yarn installs based on that, etc.

**Comment by bamader** [comment]
Looks like the only remaining thing that's failing are snapshot tests--almost there!

**Comment by lukenewman** [comment]
so weird i already did `yarn test -u`

**Comment by bamader** [comment]
I would guess pulling in base and re-pinning component library brought in some back-end stylistic things that we can't see but that are technically different in rendering terms.

**Comment by bamader** [comment]
Imma validate this bad boi now

**Comment by bamader** [comment]
## Validation Plan
- [ ] Progress indicator removed from upload page
- [ ] Back to roster button present
- [ ] Designs generally match new specs, including image
- [ ] User can upload correct file without issue
- [ ] User can upload file of wrong type (not a csv) and it's handled elegantly
- [ ] User can upload incomplete/missing info file and it's handled elegantly

**Comment by bamader** [comment]
Validations complete. I think the only thing off is the margin around the header text on the upload page itself ("Upload your Enrollment Data"). Looks like it's currently set to `margin-top-6` and `margin-bottom-4`, but comparing this to the rest of the app I think that breaks with our existing format. Elsewhere on pages with this kind of header layout, we put `margin-top-4` in the div that has the `grid-container` containing the header. The header itself we give `margin-top-0` so that it doesn't overly stack spacing. Neither one gets any `margin-bottom` styling. And then below that, the default `<p>` tag has a top margin of 1  which is a clean line break. Not a blocker by any means but it was pretty noticeable to me so wanted to point it out!

**Comment by lukenewman** [comment]
@bamader i think the difference here is that the "Back to Roster" button is inside the `grid-container`, so we need to add that margin in between that and the `h4`. i played around with those guidelines you outlined but couldn't get it right without adding in some margins

**Comment by bamader** [comment]
Gotcha, that makes sense. How does it look if we change the `margin-top-6` to a `top-4` and the `margin-bottom-4` to a `bottom-2`? Just trying to artificially create the same aesthetic as on other pages. If these number changes don't help then we can just use it as is.

**Comment by lukenewman** [comment]
Screen Shot 2021-10-01 at 4 07 38 PM

**Comment by lukenewman** [comment]
now i'm thinking the icon should be sized-down a bit

**Comment by bamader** [comment]
Yup agreed! That spacing looks good to me so let's shrink the image a bit.

**Comment by lukenewman** [comment]
eh?

Screen Shot 2021-10-01 at 4 23 36 PM

**Comment by bamader** [comment]
That looks good to me! @JordanGuinn Any thoughts on the various design screenshots?

**Comment by JordanGuinn** [comment]
Looks perfect to me! 👍 👍

**Comment by bamader** [comment]
Great! Sounds like with the new design updates, this should be good to go!

**Pull request merged**